app/models/spree/subscription_frequency.rb

```diff
@@ -1,15 +1,16 @@
 module Spree
   class SubscriptionFrequency < Spree::Base
+    include ApplicationHelper
 
     has_many :product_subscription_frequencies, class_name: "Spree::ProductSubscriptionFrequency",
                                                 dependent: :destroy
     has_many :subscriptions, class_name: "Spree::Subscription", dependent: :restrict_with_error
 
     validates :title, :months_count, presence: true
-    with_options allow_blank: true do
-      validates :months_count, numericality: { greater_than: 0, only_integer: true }
-      validates :title, uniqueness: { case_sensitive: false }
-    end
+    validates :months_count, numericality: { greater_than: 0, only_integer: true, less_than_or_equal_to: 12 }, allow_blank: true
 
+    def display_frequency
+      "#{get_title(months_count)}"
```

Comment:
you can just check if title is present then return the value and if its blank just return something like 'N/A'

Doesn't make sense to include view helper in model.

Reply:
removed allow blank true its important for stripe subscription. Made helper method to provide better details for admin, will remove this now

Reply:
even if you removed allowed blank = true now, there can still be old data which have blank value and for those we will view bad data display.
still its better to have that check as validation can be added in future again